internal/neofs/neofs.go

```diff
@@ -134,18 +133,18 @@ func (x *NeoFS) CreateContainer(ctx context.Context, prm layer.PrmContainerCreat
 	if hhDisabled, err := isHomomorphicHashDisabled(ctx, x.pool); err != nil {
 		return nil, fmt.Errorf("check homomorphic hash enabled: %w", err)
 	} else if hhDisabled {
-		cnrOptions = append(cnrOptions, container.WithAttribute(
-			"__NEOFS__DISABLE_HOMOMORPHIC_HASHING", "true"))
+		container.DisableHomomorphicHashing(&cnr)
 	}
 
-	cnr := container.New(cnrOptions...)
-
 	if prm.Name != "" {
-		container.SetNativeName(cnr, prm.Name)
+		var d container.Domain
+		d.SetName(prm.Name)
+
+		container.WriteDomain(&cnr, d)
```

Comment:
Let's move this block to previous `if prm.Name != ""` condition block in line 122